tests/kernel/workq/work_queue/src/start_stop.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2024 Måns Ansgariusson <mansgariusson@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+
+#define NUM_TEST_ROUNDS 5
+#define NUM_TEST_ITEMS  10
+/* Each work item takes 100ms by default. */
+#define WORK_ITEM_WAIT  (CONFIG_TEST_WORK_ITEM_WAIT_MS)
+/* In fact, each work item could take up to this value */
+#define WORK_ITEM_WAIT_ALIGNED                                                                     \
+	k_ticks_to_ms_floor64(k_ms_to_ticks_ceil32(WORK_ITEM_WAIT) + _TICK_ALIGN)
+#define SUBMIT_WAIT (CONFIG_TEST_SUBMIT_WAIT_MS)
+#define STACK_SIZE  (1024 + CONFIG_TEST_EXTRA_STACK_SIZE)
+#define CHECK_WAIT  ((NUM_TEST_ITEMS + 1) * WORK_ITEM_WAIT_ALIGNED)
+
+static struct k_work_q work_q;
+static K_THREAD_STACK_DEFINE(work_q_stack, STACK_SIZE);
+
+static atomic_t num_work_executed;
+static void work_handler(struct k_work *work)
+{
+	atomic_inc(&num_work_executed);
+	k_free(work);
```

Comment:
Can't do this: the k_work object is (I'm like 70% certain, and if not it absolutely should be expected to be unless we explicitly document that the pointer is now unused) still in use inside the queue code that called the handler and you're creating a free memory bug.

Really heap usage in Zephyr tests is discouraged (except obviously in heap tests, heh).  It's not the native idiom, and test code tends to rely heavily on static allocations.  That's not a firm rule, but in this case I don't see any particular value the malloc/free are bringing.

Reply:
I was worried about just this haha. 

I'll get it sorted to static memory!